pkg/kubelet/pod_workers.go

```diff
@@ -299,13 +319,82 @@ func (s *podSyncStatus) IsTerminated() bool           { return !s.terminatedAt.I
 func (s *podSyncStatus) IsFinished() bool             { return s.finished }
 func (s *podSyncStatus) IsEvicted() bool              { return s.evicted }
 func (s *podSyncStatus) IsDeleted() bool              { return s.deleted }
+func (s *podSyncStatus) IsStarted() bool              { return !s.startedAt.IsZero() }
+
+// WorkType returns this pods' current state of the pod in pod lifecycle state machine.
+func (s *podSyncStatus) WorkType() PodWorkType {
+	if s.IsTerminated() {
+		return TerminatedPodWork
+	}
+	if s.IsTerminationRequested() {
+		return TerminatingPodWork
+	}
+	return SyncPodWork
+}
+
+// replayLastUpdate creates a pendingUpdate for this status if none is currently requested
+// that matches the last update sent. If an update cannot be replayed because we do not have
+// a last update, this method is a no-op and it is the caller's responsibility to ensure
+// the worker goroutine exits gracefully.
+func (s *podSyncStatus) replayLastUpdate() {
+	if s.pendingUpdate != nil || s.lastUpdate == nil {
+		return
+	}
+	copied := *s.lastUpdate
+	s.pendingUpdate = &copied
+}
+
+// mergeLastUpdate records the most recent state from a new update. Pod and MirrorPod are
+// incremented. KillPodOptions is accumulated. If RunningPod is set, Pod is synthetic and
+// will *not* be used as the last pod state unless no previous pod state exists (because
+// the pod worker may be responsible for terminating a pod from a previous run of the
+// kubelet where no config state is visible). The contents of lastUpdate are used as the
+// source of truth for components downstream of the pod workers.
+func (s *podSyncStatus) mergeLastUpdate(other UpdatePodOptions) {
+	opts := s.lastUpdate
+	if opts == nil {
+		opts = &UpdatePodOptions{}
```

Comment:
Do we need to copy over / initialize `opts.StartTime`? Otherwise it will be lost I think...

Reply:
thank you, that is a bigger todo.  start time needs to be tested independently cause i ripped some code out and moved it, and probably is broken (the metric).